File: exercises/easy-level/leet-code-problems/src/main/java/org/example/solutions/DecodeXORArray.java
package org.example.solutions;

public class DecodeXORArray {
  public void testProblem() {
    int[] encoded = {1,2,3};
    int[] response = this.decode(encoded, 1);
    for (int j : response) System.out.print(j);
  }

  public int[] decode(int[] encoded, int first) {
    int[] resp = new int[encoded.length + 1];
    resp[0] = first;
    for (int i = 0; i < encoded.length; i++)
      resp[i + 1] = resp[i] ^ encoded[i];
    return resp;
  }
}

File: exercises/easy-level/decodeXORarray.cpp
// 1720. Decode XORed Array
#include <iostream>
#include <vector>

using namespace std;

vector<int> decode(vector<int> &encoded, int first)
{
  vector<int> resp;
  resp.push_back(first);

  for (int i = 0; i < encoded.size(); i++)
    resp.push_back(resp[i] ^ encoded[i]);

  return resp;
}

int main()
{
  vector<int> encoded = {1, 2, 3};
  vector<int> resonse = decode(encoded, 1);

  for (int i = 0; i < resonse.size(); i++)
    cout << resonse[i];
  cout << endl;

  return 0;
}